### PR TITLE
Fix #5017 pt2 - swings and merry-go-rounds

### LINKED ISSUE #5017: ClinVar CLNSIG conflicting pathogenicity parsing
CLNSIG “Conflicting_interpretations_of_pathogenicity” has changed into “Conflicting_classifications_of_pathogenicity” sometime after the 20231203 dl, before the 240215.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

Fortunately, due to the sloppy info string match, this was never an issue. Adding the proper strings and numerical values for good measure. We could also make it a regex or more strict string check here first: thoughts?
I would tentatively argue to leave it like this until we mend the same issue in genmod.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

File: scout/parse/variant/clnsig.py [hunk 1]
```diff
@@ -100,20 +100,23 @@ def is_pathogenic(variant):
...
+    load_categories = {
+        "pathogenic",
+        "likely_pathogenic",
+        "conflicting_classifications_of_pathogenicity",
+        "conflicting_interpretations_of_pathogenicity",
+        "conflicting_interpretations",
+    }
+
     # Otherwise check if clinvar pathogenicity status is in INFO field
     clnsig_accessions = parse_clnsig(variant)
```
Comment (northwestwitch) [inline scout/parse/variant/clnsig.py:121]: It could help to add a type hint to the variable

File: scout/parse/variant/clnsig.py [hunk 2]
```diff
@@ -122,7 +125,7 @@ def is_pathogenic(variant):
         if clnsig in load_categories:
             return True
         if isinstance(clnsig, int):
-            if clnsig == 4 or clnsig == 5:
+            if clnsig == 4 or clnsig == 5 or clnsig == 8:
```
Comment (northwestwitch) [inline scout/parse/variant/clnsig.py:128]:
```
if clnsig in load_categories or clnsig in [4, 5, 8]:
            return True
```

**Review by northwestwitch — APPROVED** [review]
Looks good to me. Good catch! 👍🏻

File: scout/parse/variant/clnsig.py [hunk 3]
```diff
@@ -122,7 +125,7 @@ def is_pathogenic(variant):
...
                 return True
```
Comment (northwestwitch) [inline scout/parse/variant/clnsig.py:129]:
```
if clnsig in load_categories or clnsig in [4, 5, 8]:
            return True
```

File: scout/parse/variant/clnsig.py (hunk 1 above)
Comment (dnil) [reply scout/parse/variant/clnsig.py:121]: Eh, that should always be apparent on declaration, I hope? You mean one could mistake it for a dict? Why not, yes - it's not so commonly used I guess. Or in the def statement? I'll fix that as well to be sure.

File: scout/parse/variant/clnsig.py (hunk 3 above)
Comment (dnil) [reply scout/parse/variant/clnsig.py]: Right, that will actually work. I'll put them all in the same set!

File: scout/parse/variant/clnsig.py [hunk 4]
```diff
@@ -87,42 +87,47 @@ def parse_clnsig(variant, transcripts=None):
...
+        8,
+    }
+
     # Otherwise check if clinvar pathogenicity status is in INFO field
     clnsig_accessions = parse_clnsig(variant)
 
     for annotation in clnsig_accessions:
         clnsig = annotation["value"]
         if clnsig in load_categories:
             return True
```
Comment (northwestwitch) [inline scout/parse/variant/clnsig.py:132]: Yes, easier!

File: scout/parse/variant/clnsig.py (hunk 3 above)
Comment (dnil) [reply scout/parse/variant/clnsig.py]: Loose typing is nice! 😁

**Comment by dnil** [comment]
Hm, what happened to the lint-fixing?

File: scout/parse/variant/clnsig.py (hunk 1 above)
Comment (dnil) [reply scout/parse/variant/clnsig.py:126]: Ah, I hear you! I'll add a return type to `parse_clnsig` instead - should solve it for most editors!

**Pull request merged**